### DISCUSSION TITLE
A question about the "configuration" in the tensor mechanics

### DISCUSSION THREAD

**Discussion by xuxiaobei1995** [idaholab/moose#23045] [Q&A Modules: Solid mechanics] [answered]
Hi all,

Recently I was reading the code in the tensor mechanics. I was confused about the "configuration" in the code. For example, the line 114-120 in ADComputeFiniteStrain.C are:

```
// Update strain in intermediate configuration
  _mechanical_strain[_qp] = _mechanical_strain_old[_qp] + _strain_increment[_qp];
  _total_strain[_qp] = _total_strain_old[_qp] + total_strain_increment;

  // Rotate strain to current configuration
  _mechanical_strain[_qp].rotate(_rotation_increment[_qp]);
  _total_strain[_qp].rotate(_rotation_increment[_qp]);
```

What do the "intermediate configuration" and the "current configuration" mean? Why should the mechanical and total strain be "rotated" but the strain increment don't need to? Does any one know this? Thanks.

Best Regards,
Xiaobei

**Comment by hugary1995** [comment, marked as answer]
A general motion with geometric nonlinearity has many configurations which are essentially different ways of observing the same motion. Different choice of configurations leads to different (time discrete) constitutive update of the stress-strain relation. Some may be more convenient than others. You can read "Computational Inelasticity" by Simo and Hughes for more details.

The one implemented in `ComputeFiniteStrain` is a special one with a corotational intermediate configuration. See the paper by Rashid for details: https://doi.org/10.1002/nme.1620362302

**Reply by xuxiaobei1995** [reply]
Hi Gary, thanks for your guidance! I will read the literatures you mentioned to learn that.
